### PR TITLE
Use Cirrus CI to test FreeBSD

### PR DESCRIPTION
**Pull request by kiwiroy**
Enable integration testing of FreeBSD in addition to linux and macos on travis.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+task:
+  freebsd_instance:
+    matrix:
+      - image: freebsd-11-2-release-amd64
+      - image: freebsd-12-0-release-amd64
+  install_script: pkg install -y gmake
+  script:
+    - gmake all test_programs && cat benchmark | ./benchmark &&  ./test_slow_decompression
```
Comment (fkorotkov) [inline .cirrus.yml]: Maybe improve readability?

```suggestion
    - gmake all test_programs
    - cat benchmark | ./benchmark
    - ./test_slow_decompression
```

File: .cirrus.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+task:
```
Comment (fkorotkov) [inline .cirrus.yml]: Maybe name the task?

```suggestion
Test_task:
```

Can also be achieved by adding `name: Test` under `task`.

**Comment by fkorotkov** [comment]
BTW @ebiggers will need to install [Cirrus CI](https://github.com/marketplace/cirrus-ci) for this repo.

**Comment by ebiggers** [comment]
Merged via https://github.com/ebiggers/libdeflate/pull/54.  Thanks!

**Pull request closed without merging**